src/main/java/picard/illumina/NewIlluminaBasecallsConverter.java

```diff
@@ -161,10 +161,21 @@ public void doTileProcessing() {
         tileProcessingExecutor.shutdown();
 
         awaitThreadPoolTermination("Reading executor", tileProcessingExecutor);
-        awaitThreadPoolTermination("Tile completion executor", completedWorkExecutor);
 
-        barcodeWriterThreads.values().forEach(ThreadPoolExecutor::shutdown);
-        barcodeWriterThreads.forEach((barcode, executor) -> awaitThreadPoolTermination(barcode + " writer", executor));
+        // if there was an exception reading then purge the completion and writer queues and initiate shutdown.
+        if(tileProcessingExecutor.exception != null) {
+            completedWorkExecutor.purge();
+            barcodeWriterThreads.values().forEach(ThreadPoolExecutor::purge);
+
+            List<Runnable> tasksStillRunning = new ArrayList<>(completedWorkExecutor.shutdownNow());
+            barcodeWriterThreads.values().forEach(executor -> tasksStillRunning.addAll(executor.shutdownNow()));
+            throw new PicardException("Reading executor had exceptions " + tasksStillRunning.size()
```

Comment:
missing a `.` after `exceptions`? Might be clearer as `... had exceptions. There were " + size() +" tasks still running or queued ...`